web/src/ui.rs

```diff
@@ -97,4 +206,49 @@ impl UiBackend for WebUiBackend {
     fn message(&self, message: &str) {
         self.js_player.display_message(message);
     }
+
+    fn display_file_open_dialog(&mut self, filters: Vec<FileFilter>) -> Option<DialogResultFuture> {
+        // Prevent opening multiple dialogs at the same time
+        if self.dialog_open {
+            return None;
+        }
+        self.dialog_open = true;
+
+        // Create the dialog future
+        Some(Box::pin(async move {
+            let mut dialog = AsyncFileDialog::new();
+
+            for filter in filters {
+                if std::env::consts::OS == "macos" && filter.mac_type.is_some() {
```

Comment:
I'm pretty sure `std::env::consts::OS` is some other string on Web, so this condition would never hold.

Reply:
Nice catch, on web this is apparently an empty string. Changed this to use `window.navigator.platform`